File: app/src/main/java/xyz/klinker/messenger/view/AutoSummaryEditTextPreference.java
package xyz.klinker.messenger.view;

import android.content.Context;
import android.preference.EditTextPreference;
import android.util.AttributeSet;

public class AutoSummaryEditTextPreference extends EditTextPreference {
    public AutoSummaryEditTextPreference(Context context) {
        this(context, null);
    }

    public AutoSummaryEditTextPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onDialogClosed(boolean positiveResult) {
        super.onDialogClosed(positiveResult);
        if (positiveResult) {
            setSummary(getSummary());
        }
    }

    @Override
    public CharSequence getSummary() {
        return getText();
    }
}

File: shared/src/main/java/xyz/klinker/messenger/shared/view/AutoSummaryEditTextPreference.kt
package xyz.klinker.messenger.shared.view

import android.content.Context
import android.preference.EditTextPreference
import android.util.AttributeSet

class AutoSummaryEditTextPreference : EditTextPreference {

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    constructor(context: Context) : super(context)

    override fun onDialogClosed(positiveResult: Boolean) {
        super.onDialogClosed(positiveResult)
        if (positiveResult) {
            summary = summary
        }
    }

    override fun getSummary(): CharSequence? {
        return text
    }
}
